services/blockchain-indexer/shared/dataService/business/events.js

```diff
@@ -176,42 +196,43 @@ const getEvents = async params => {
 		const { topic, ...remParams } = params;
 		params = remParams;
 
-		const topics = topic.split(',');
-		const numUniqueTopics = dropDuplicates(topics).length;
-		topics.forEach(t => {
-			if (t.length === LENGTH_ID) {
-				topics.push(EVENT_TOPIC_PREFIX.TX_ID.concat(t), EVENT_TOPIC_PREFIX.CCM_ID.concat(t));
-			} else if (
-				t.startsWith(EVENT_TOPIC_PREFIX.TX_ID) &&
-				t.length === EVENT_TOPIC_PREFIX.TX_ID.length + LENGTH_ID
-			) {
-				// Check for the transaction ID both with and without the topic prefix
-				topics.push(t.slice(EVENT_TOPIC_PREFIX.TX_ID.length));
-			} else if (
-				t.startsWith(EVENT_TOPIC_PREFIX.CCM_ID) &&
-				t.length === EVENT_TOPIC_PREFIX.CCM_ID.length + LENGTH_ID
-			) {
-				// Check for CCM ID both with and without the topic prefix
-				topics.push(t.slice(EVENT_TOPIC_PREFIX.CCM_ID.length));
-			}
+		const allTopics = topic.split(',');
+		if (topic.length === LENGTH_ID) {
```

Comment:
This entire if..else ladder needs to be within a for loop against the allTopics.